test/e2e/scheduling/priorities.go

```diff
@@ -254,6 +267,37 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(tolePod.Spec.NodeName).To(Equal(nodeName))
 	})
+	It("Pod should be preferably scheduled to nodes which satisfy its limits", func() {
+		// Update one node to have large allocatable.
+		nodeName := nodeList.Items[0].Name
+		err := UpdateMemoryOfNode(cs, nodeName, int64(10000))
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			nodeOriginalMemory, found := nodeList.Items[0].Status.Allocatable[v1.ResourceMemory]
```

Comment:
Shouldn't you read the original amount of memory at the beginning of the function, before you update it?